rllib/examples/documentation/saving_and_loading_algos_and_policies.py

```diff
@@ -0,0 +1,308 @@
+# flake8: noqa
+
+# __create-algo-checkpoint-begin__
```

Comment:
By convention (and I think this is a good one to keep consistent), we have a `doc_code` folder like here for Tune:

https://github.com/ray-project/ray/tree/master/doc/source/tune/doc_code

which has all the code referenced in docs. You can then basically copy this block in the bazel `BUILD` file to auto-test all imported snippets. I'm aware that this would likely also happen with `rllib/examples` but I think it'd be great to keep this code close to doc-sources (which e.g. makes it likely easier for new contributors to understand the structure).


https://github.com/ray-project/ray/blob/master/doc/BUILD#L155-L164

Reply:
Should be a simple move... apart from that, this is exactly what I had in mind! :D

Reply:
Dumb question on this one:
If we currently run all rllib/examples/documentation/* via the CI, how would we reference into the new location (docs/source/rllib/doc_code/*) from the rllib/BUILD file?

Reply:
we should simply consider those two different CI runs. I didn't realise there was an `rllib/examples/documentation` folder already. We could either leave it as is, or (in a follow-up PR) move the contents of that folder into `doc/source/rllib/doc_code`. We can then delete that part from the `rllib/BUILD` and add the resp. bit to `doc/BUILD`. that way we don't have to think about the issue you brought up. wdyt?

Reply:
let's simply skip this discussion for now and fix this later, don't think it's a big deal either way